src/feature_engineer.py

```diff
@@ -56,6 +65,99 @@ def feature_engineer_data():
     print("\n##### feature_engineer: Finished processing features!")
 
 
+def build_common_name_list():
+    """
+    Returns a table of common names.
+
+    Parameters:
+    ------
+
+    Returns:
+    -------
+    table of common names: DataFrame
+
+    """
+    # URLs with common names from census for male and female
+    url_male = "https://namecensus.com/male_names_alpha.htm"
+    url_female = "https://namecensus.com/female_names_alpha.htm"
+
+    male_names_page_1 = requests.get(url_male)
+    top_male_first_name = pd.read_html(male_names_page_1.text)
+
+    female_names_page_1 = requests.get(url_female)
+    top_female_first_name = pd.read_html(female_names_page_1.text)
+
+    top_names = pd.concat([top_male_first_name[0], top_female_first_name[0]], ignore_index=True)
+    return pd.DataFrame(top_names["Name"].str.lower())
+
+def get_name_length(name):
+    """
+    Returns the length of first name.
+
+    Parameters:
+    ------
+    name: (str)
+    the input name
+
+    Returns:
+    -------
+    length of name: (float)
+
+    """
+    if name == name:
+        return len(name)
+    else:
+        return 0
+
+def has_last_name(name):
+    """
+    Returns whether the character has a last name.
+    
+    Parameters:
+    ------
+    name: (str)
+    the input name
+    
+    Returns:
+    ------
+    True/False of whether the last name exists (boolean)
+    
+    """
+    return name == name
+    
+def norm(appearances, year, MAX_YR=2013):
```

Comment:
nice, I though we may need to take another year, that way it would be easy to get that